c_src/ep_encoder.c

```diff
@@ -515,9 +515,11 @@ pack_bin_as_int64(ErlNifEnv *env, ERL_NIF_TERM term, ep_enc_t *enc)
     int64_t     val;
     val = to_integer(bin.data, bin.size);
 
-    if (enc->omit && val == 0) {
-        return RET_OK;
-    }
+
+    if (val == 0 && field->proto_v == 3 && field->o_type == occurrence_defaulty) {
+         return RET_OK;
+     }
+
```

Comment:
why change this?

Reply:
`enc->omit` was removed with the proto3 changes upstream; this is what they've changed this idiom to elsewhere in the file and this behavior seems correct to me.

Reply:
did you get a chance to check if it works fine with the protobufs defined in our repo.
you can test some protobuf definitions in this repo standalone.. were you able to test our protobufs?

Reply:
ok looked into it and since `proto_v` is set to `2` by default and `enc->omit` was also only set if `proto_v` was `3`, i don't think this case wasn't going off before, so this shouldn't change anything. i'm still going to test this case to make sure it works the same as on master though.

Reply:
```
-record(m1, {a}).

encode_msg(Msg, Defs) ->
    encode_msg(Msg, Defs, [{string_as_list, true}, {with_utf8, true}]).

encode_msg(Msg, Defs, Opts) ->
    enif_protobuf:set_opts(Opts),
    enif_protobuf:load_cache(Defs),
    enif_protobuf:encode(Msg).


encodes_binary_correctly_test() ->
    <<8, 0>> = encode_msg( #m1{a = <<>>},
        [{{msg, m1}, [#?gpb_field{name = a, fnum = 1, rnum = #m1.a,
                type = int64, occurrence = optional,
                opts = [{ebin, true}]}]}]
        ).
```
This snippet passes on both `master` and this branch so I'm pretty confident this change isn't breaking anything.

Reply:
i would use our own protobuf definitions - are you sure the proto_v is being picked up properly from our protobuf files?
the default occurrence type is optional for proto3.